x-pack/legacy/plugins/code/index.ts

```diff
@@ -44,6 +45,12 @@ export const code = (kibana: any) =>
         };
       },
       hacks: ['plugins/code/hacks/toggle_app_link_in_nav'],
+      savedObjectSchemas: {
+        [SAVED_OBJ_REPO_REF]: {
+          isNamespaceAgnostic: true,
```

Comment:
I think `isNamespaceAgnostic` needs to be false (or omitted). As-is the saved object will be treated as global instead of space aware, even though namespace is being set.  It will cause side effects like not removing these saved objects if the entire space is deleted.

Reply:
Disregard my last comment - now that I see the repository reference helper,  I see this is meant to be space agnostic object with a list of namespaces.